src/wallet/wallet.cpp

```diff
@@ -4135,7 +4188,26 @@ LegacyScriptPubKeyMan* CWallet::GetOrCreateLegacyScriptPubKeyMan()
 
 void CWallet::SetupLegacyScriptPubKeyMan()
 {
-    if (!m_spk_man) m_spk_man = MakeUnique<LegacyScriptPubKeyMan>(*this);
+    if (!m_internal_spk_managers.empty() || !m_external_spk_managers.empty() || !m_spk_managers.empty()) {
+        return;
+    }
+
+    assert(m_internal_spk_managers.empty() && m_external_spk_managers.empty() && m_spk_managers.empty());
```

Comment:
This assert seems a little unnecessary considering the if-statement above

Reply:
Removed